addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/handler/UniFiClientHandler.java

```diff
@@ -0,0 +1,215 @@
+import static org.eclipse.smarthome.core.types.RefreshType.REFRESH;
+import static org.openhab.binding.unifi.UniFiBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.unifi.UniFiBindingConstants;
+import org.openhab.binding.unifi.internal.UniFiClientConfig;
+import org.openhab.binding.unifi.internal.UniFiController;
+import org.openhab.binding.unifi.internal.api.UniFiDevice;
+import org.openhab.binding.unifi.internal.api.UniFiSite;
+import org.openhab.binding.unifi.internal.api.UniFiWirelessClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link UniFiClientHandler} is responsible for handling commands and status
+ * updates for UniFi Wireless Devices.
+ *
+ * @author Matthew Bowman - Initial contribution
+ */
+public class UniFiClientHandler extends BaseThingHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
+            .singleton(UniFiBindingConstants.THING_TYPE_CLIENT);
+
+    private Logger logger = LoggerFactory.getLogger(UniFiClientHandler.class);
+
+    private volatile UniFiClientConfig config; /* mgb: volatile because accessed from multiple threads */
+
+    public UniFiClientHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        // mgb: called when the config changes
+
+        config = getConfig().as(UniFiClientConfig.class).tidy();
+        logger.debug("Initializing the UniFi Client Handler with config = {}", config);
+
+        if (!config.isValid()) {
+            updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid MAC address.");
+            return;
+        }
+
+        Bridge bridge = getBridge();
+
+        if (bridge == null || bridge.getHandler() == null || !(bridge.getHandler() instanceof UniFiControllerHandler)) {
+            updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "You must choose a UniFi Controller for this UniFi Wireless Device.");
+            return;
+        }
+
+        if (bridge.getStatus() == OFFLINE) {
```

Comment:
I'm not sure what you mean. At this point in the code, the client is being configured and if the controller (bridge) isn't online yet, does it make sense for the client to be marked as online?

I did it this way because I saw `ThingStatusDetail.BRIDGE_OFFLINE` and thought this was exactly its use case.